datafusion/optimizer/src/utils.rs

```diff
@@ -583,6 +583,7 @@ where
 fn name_for_alias(expr: &Expr) -> Result<String> {
     match expr {
         Expr::Sort { expr, .. } => name_for_alias(expr),
+        Expr::Wildcard => Ok("*".to_string()),
         expr => expr.display_name(),
```

Comment:
As you note in your PR description, another way to fix this would be to support `Expr::Wildcard` in `Expr::display_name`

Unless there is some reason to not support `WildCard`s in `Expr::display_name` I would personally prefer adding the support there. It seems strange to put code for Wildcard in a function that seem to be trying to unwrap `Expr::Sort`s 🤔 

https://github.com/apache/arrow-datafusion/blob/fddb3d3651041f41d66a801f10e27387e84374f7/datafusion/expr/src/expr.rs#L1358-L1360

Reply:
Thank you very much @alamb! I agree with you, so I've just moved that line to the `create_name` function called by `Expr::display_name`. If this is fine for you, I guess we can merge it.